kubechain/internal/controller/taskruntoolcall/taskruntoolcall_controller.go

```diff
@@ -646,9 +669,29 @@ func (r *TaskRunToolCallReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return ctrl.Result{}, nil
 	}
 
-	// Step 3: Check if this is an MCP tool
 	serverName, mcpToolName, isMCP := isMCPTool(trtc.Spec.ToolRef.Name)
 
+	if isMCP {
+		// Step 3: Check if this is an MCP tool and needs approval
+		mcpServer, needsApproval, err := r.getMCPServer(ctx, &trtc)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if needsApproval {
+			trtc.Status.Status = "AwaitingHumanApproval"
+			trtc.Status.StatusDetail = fmt.Sprintf("Waiting for human approval via contact channel %s", mcpServer.Spec.ApprovalContactChannel.Name)
```

Comment:
- fetch the contact channel based on the MCP server ref
- create a post /humanlayer/v1/function_calls
- store the call ID on the status
- requeue in 5 seconds
- in reconcile loop

```
if status == AwaitingHumanApproval {
   GET /humanalyer/v1/function_calls/{call_id}
   if not finished:
        requeue in 5s
   else -> either execute tool, or set output to denied
```